calico_containers/pycalico/netns.py

```diff
@@ -180,6 +179,32 @@ def remove_ip_from_ns_veth(container_pid, ip, veth_name_ns):
                        "dev", "%(device)s" % veth_name_ns])
 
 
+def provision_veth(self, endpoint, ns_pid, veth_name_ns):
+    """
+    Create the veth, move into the container namespace, add the IP and
+    set up the default routes.
+
+    Note, the endpoint will not be updated in ETCD. If desired, the user
+    should update the endpoint mac with the mac address provided
+    by the function and then call update_endpoint
+
+    :param endpoint: The endpoint object to provision the veth on
+    :param ns_pid: The PID of the namespace to operate in
+    :param veth_name_ns: The name of the interface in the namespace
+    :return The mac address of the veth as a string
+    """
+    create_veth(endpoint.name, endpoint.temp_interface_name)
+    for ip in endpoint.ipv4_nets | endpoint.ipv6_nets:
```

Comment:
Yep. You should probably add a comment in create_endpoint explaining why
you're creating a network since it's non-obvious.

Bonus points for adding a comment to datastore.py (perhaps with Spike's
explanation); I can't find an explanation for the translation to a network
anywhere in the code.

On Mon, Aug 10, 2015 at 4:47 PM, Luke Mino-Altherr <notifications@github.com
> wrote:

> In calico_containers/pycalico/netns.py
> <https://github.com/projectcalico/libcalico/pull/8#discussion_r36698318>:
>
> > +def provision_veth(self, endpoint, ns_pid, veth_name_ns):
> > +    """
> > +    Create the veth, move into the container namespace, add the IP and
> > +    set up the default routes.
> > +
> > +    Note, the endpoint will not be updated in ETCD. If desired, the user
> > +    should update the endpoint mac with the mac address provided
> > +    by the function and then call update_endpoint
> > +
> > +    :param endpoint: The endpoint object to provision the veth on
> > +    :param ns_pid: The PID of the namespace to operate in
> > +    :param veth_name_ns: The name of the interface in the namespace
> > +    :return The mac address of the veth as a string
> > +    """
> > +    create_veth(endpoint.name, endpoint.temp_interface_name)
> > +    for ip in endpoint.ipv4_nets | endpoint.ipv6_nets:
>
> @Symmetric <https://github.com/Symmetric> Do you agree with Spike? I
> fixed the way I am accessing the IP addresses (indicating .ip on the
> IPNetwork obj), however, I am still pulling the IPs from the endpoint's
> ipvX_nets.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/projectcalico/libcalico/pull/8/files#r36698318>.
>